### PR TITLE
Added support for tailing error logs 

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation
Allow to tail both error and stdout in one command. 
Solution is using list to fetch all logs and then selecting two most recent ones. 
This is workaround for current api limitations. 

Code inspired on another PR added by community member. 

# Side effects

Tailing 2 different files so logs can appear in different order 

ping @alanmoran, @pb82, @paolobueno, @ziccardi

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

File: lib/cmd/fh3/app/logs/tail.js [hunk 1]
```diff
@@ -32,20 +33,60 @@ module.exports = {
...
+    if (params.last) {
+      stdRequest.last = params.last;
+    }
+    if (params.offset) {
+      stdRequest.offset = params.offset;
+    }
+    if (params.logname) {
+      stdRequest.logname = params.logname;
+      return cb(null, stdRequest, null);
+    }else{
```
Comment (wei-lee) [inline lib/cmd/fh3/app/logs/tail.js]: perhaps move the statements in the else block to its own function

File: lib/cmd/fh3/app/logs/tail.js [hunk 2]
```diff
@@ -32,20 +33,60 @@ module.exports = {
...
+          var stdLog = self.getFirst(res.logs, "stdout.log.");
+          var errorLog = self.getFirst(res.logs, "stderr.log.");
+          var errorRequest = _.clone(stdRequest);
+          stdRequest.logname = stdLog.name;
+          errorRequest.logname = errorLog.name;
+          log.warn("No log file specified. Fetching logs from latest log files. \nStandard output: " +
+            stdLog.name + "\nError log: " + errorLog.name);
+          return cb(null, stdRequest, errorRequest);
+        }else{
+          return cb(null, stdRequest, null);
```
Comment (wei-lee) [inline lib/cmd/fh3/app/logs/tail.js]: so there is no log file in the else statement here, is it by default just return the default latest log file?

File: lib/cmd/fh3/app/logs/tail.js [hunk 3]
```diff
@@ -32,20 +33,60 @@ module.exports = {
...
+        }
+      });
+    }
+  },
+  'getFirst': function(list,name){
+    return _.find(list, function(element){
```
Comment (wei-lee) [inline lib/cmd/fh3/app/logs/tail.js]: given the name of the function, probably here should use _.first?

File: lib/cmd/fh3/app/logs/tail.js (hunk 2 above)
Comment (wtrocki) [inline lib/cmd/fh3/app/logs/tail.js]: If there is error or somehow we cannot obtain list of logs we just defaulting to old behavior.
I was thinking about adding some log.warn here as well. What you think?

Comment (wei-lee) [inline lib/cmd/fh3/app/logs/tail.js]: I think so. The _err_ from list endpoint is not handled here. Perhaps just fail the command if there is an error when listing logs.

File: lib/cmd/fh3/app/logs/tail.js (hunk 3 above)
Comment (wtrocki) [inline lib/cmd/fh3/app/logs/tail.js]: First will accept only number of items - no predicate possible.

Comment (wei-lee) [inline lib/cmd/fh3/app/logs/tail.js]: ah, ok. Me being stupid...

**Comment by ziccardi** [comment]
I think that tailing both STD-ERR and STD-OUT logs together this way can be really confusing: you can't know if a message is from ERR or from OUT and, moreover, they won't be chronologically ordered.

A small workaround that could just alleviate a bit the issue, could be to prepend the name of the log file before each message

**Comment by alanmoran** [comment]
@ziccardi this is a good point. The consulting team use logging library so we can clearly see where logs are coming.

@wtrocki :+1: for this PR. The consulting team thank you!

**Comment by pb82** [comment]
LGTM, but one question: do you think it would be a good idea to make this behaviour explicit by forcing it through a new flag (like --mixedLogs)?

**Comment by wtrocki** [comment]
I like idea of mixedLogs but it may be just noise in this particular case. Users may expect stdout and err as default behavior: Just tail me all app logs and that's it. This may be used in most of the cases. If different handling is required we can always specify file name. New flag may be just missed by user.

File: lib/cmd/fh3/app/logs/tail.js [hunk 4]
```diff
@@ -55,11 +101,12 @@ module.exports = {
         }
         if (msg.data && msg.data.length !== 0) {
           var logString = msg.data.join("\n");
-          console.log(logString);
+          console.log(params.logname + ":");
+          console.log(logString+"\n");
```
Comment (fh-qube) [inline lib/cmd/fh3/app/logs/tail.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**